docs-internal/multipart.md

```diff
@@ -0,0 +1,328 @@
+
+    static Builder builder(){
+        return new BodyPartBuilder();
+    }
+
+    static interface Builder extends io.helidon.common.Builder<BodyPart> {
+
+        <T> Builder entity(T entity);
+
+        Builder headers(BodyPartHeaders header);
+
+        Builder content(Publisher<DataChunk> content);
+
+        BodyPart build();
+    }
+}
+```
+
+```java
+public interface MultiPart extends BodyPart, Publisher<BodyPart> {
+
+    static <T> MultiPart create(T ... entities){
+        Builder builder = builder();
+        for(T entity : entities){
+            builder.bodyPart(BodyPart.create(entity));
+        }
+        return builder.build();
+    }
+
+    static MultiPart create(Publisher<BodyPart> publisher){
+        return builder().bodyParts(publisher).builder();
+    }
+
+    static MultiPart create(BodyPart ... bodyParts){
+        Builder builder = builder();
+        for(BodyPart bodyPart : bodyParts){
+            builder.bodyPart(bodyPart);
+        }
+        return builder.build();
+    }
+
+    static Builder builder(){
+        return new MultiPartBuilder();
+    }
+
+    static interface Builder extends io.helidon.common.Builder<MultiPart> {
+
+        Builder bodyPart(BodyPart bodyPart);
+
+        Builder bodyParts(BodyPart ... bodyParts);
+
+        Builder bodyParts(Publisher<BodyPart> bodyParts);
+
+        MultiPart build();
+    }
+}
+```
+
+```java
+public interface BufferedMultiPart {
+
+    Iterable<BodyPart> bodyParts();
+
+    static <T> MultiPart create(T ... entities){
+        Builder builder = builder();
+        for(T entity : entities){
+            builder.bodyPart(BodyPart.create(entity));
+        }
+        return builder.build();
+    }
+
+    static Builder builder(){
+        return new BufferedMultiPartBuilder();
+    }
+
+    static interface Builder extends MultiPart.Builder {
+
+        BufferedMultiPart build();
+    }
+}
+```
+
+```java
+public class BodyPartHeaders extends Parameters {
+
+    String name();
+
+    String filename();
+
+    String contentType();
+
+    String contentTransferEncoding();
+
+    Charset charset();
+
+    long size();
+
+    static Builder builder(){
+        return new BodyPartHeadersBuilder();
+    }
+
+    static interface Builder extends io.helidon.common.Builder<BodyPartHeaders> {
+
+        Builder name(String name);
+
+        Builder filename(String filename);
+
+        Builder contentType(String contentType);
+
+        Builder contentTransferEncoding(String contentTransferEncoding);
+
+        Builder charset(Charset charset);
+
+        Buidler size(long size);
+
+        Builder header(String name, String value);
+
+        Builder headers(Map<String, String> headers);
+
+        BodyPartHeaders build();
+    }
+}
+```
+
+```java
+public class MultiPartSupport implements Service, Handler {
+    // ...
+}
+```
+
+## Open Issues
+
+- Should we support deserializing from the top level entity, e.g. as a `List<File>` ?
+  - This requires generic type mapping
```

Comment:
Version 2.0?

Reply:
Do you mean that we can add such feature in a future release ?

Reply:
I've added a sub-section and an example for this, as well as a note about it being supported in a future release. Please confirm if that's aligned with your comment, if so I'll mark this as resolved.